src/node.cpp

```diff
@@ -1047,9 +1049,18 @@ void LocalNode::setnameparent(LocalNode* newparent, string* newlocalpath)
         // (we don't construct a UTF-8 or sname for the root path)
         parent->children[&localname] = this;
 
-        if (sync->client->fsaccess->getsname(newlocalpath, &slocalname))
+        if (!slocalname)
         {
-            parent->schildren[&slocalname] = this;
+            slocalname = new string();
+        }
+        if (sync->client->fsaccess->getsname(newlocalpath, slocalname) && slocalname->size() && *slocalname != localname)
```

Comment:
It seems that `slocalname->size()` is always true if `getsname()` returned true. Maybe the check is not needed.

Reply:
I agree. I added it basically to protect the SDK against a malfunction of `getsname` that is platform-specific.